### PR TITLE
Advanced Clown Shoes now require the user to be clumsy instead of requiring to be a clown.

### LINKED ISSUE #21312: Request: Make Advanced Clownshoes Require Clumsy
Currently they are hardcoded to only work if you have the assigned role clown. I feel that instead, it should just check for the genetic defect. This way clowns who lose their clumsiness can't use the shoes and those who earn the mighty clumsy power CAN use them.

This would be an easy change/good first issue.

### PR DESCRIPTION
**Pull request by boy2mantwicethefam**
closes #21312 
That was easy
:cl:
 * tweak: The advanced clown shoes now require the user to be clumsy instead of being a clown.

### CODE REVIEW

**Comment by Dacendeth** [comment]
Now if only clowns kept their rye immune clumsy after being cloned

**Comment by ShiftyRail** [comment]
The feature request also suggested to remove the hard-coded "clown" requirement.

**Comment by boy2mantwicethefam** [comment]
Hmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmm

**Comment by Intigracy** [comment]
You made it so people who are clumsy can't use the shoes fam.

File: code/modules/clothing/shoes/miscellaneous.dm [hunk 1]
```diff
@@ -195,7 +195,7 @@
 	var/random_sound = 0
 
 /obj/item/clothing/shoes/clown_shoes/advanced/attack_self(mob/user)
```
Comment (Intigracy) [inline code/modules/clothing/shoes/miscellaneous.dm]:
```suggestion
/obj/item/clothing/shoes/clown_shoes/advanced/attack_self(mob/living/user)
```

File: code/modules/clothing/shoes/miscellaneous.dm [hunk 2]
```diff
@@ -195,7 +195,7 @@
...
-	if(user.mind && user.mind.assigned_role != "Clown")
+	if(user.mind && M_CLUMSY in user.mutations)
```
Comment (Intigracy) [inline code/modules/clothing/shoes/miscellaneous.dm]:
```suggestion
	if(user.mind && !clumsy_check(user))
```

**Review by Intigracy — CHANGES_REQUESTED** [review]
the way you have this doesn't even work right, it works for everyone who ISN'T clumsy

**Review by Intigracy — APPROVED** [review]

**Pull request merged**